### PR TITLE
Return meaningful error on empty bucket sync for model uri

### PR DESCRIPTION
**Pull request by cliveseldon**
rclone will happily sync an empty bucket without error which results in a later error on model load that no rclone folder can be found which is not very informative to the user. This update checks if no rclone folder was created in the sync and returns an informative error.

### CODE REVIEW

File: scheduler/pkg/agent/repository/triton/config/model_config.pb.go [hunk 1]
```diff
@@ -48,11 +48,11 @@ const (
 	_ = protoimpl.EnforceVersion(protoimpl.MaxVersion - 20)
 )
 
-//@@
-//@@.. cpp:enum:: DataType
-//@@
-//@@   Data types supported for input and output tensors.
-//@@
+// @@
```
Comment (sakoush) [inline scheduler/pkg/agent/repository/triton/config/model_config.pb.go:51]: why is this file changing?

File: samples/models/error-uri-notexist.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: mlops.seldon.io/v1alpha1
```
Comment (sakoush) [inline samples/models/error-uri-notexist.yaml:1]: this is used?

**Review by sakoush — COMMENTED** [review]
I left some clarification comments.

File: scheduler/pkg/agent/rclone/rclone.go [hunk 3]
```diff
@@ -308,6 +320,14 @@ func (r *RCloneClient) Copy(modelName string, srcUri string, config []byte) (str
 	if err != nil {
 		return "", fmt.Errorf("Failed to sync/copy %s to %s %w", srcUpdated, dst, err)
 	}
+	// Even if we had success from rclone the src may be empty so need to check
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply scheduler/pkg/agent/rclone/rclone.go:323]: I think we assume at least a model-settings.json though?

File: scheduler/pkg/agent/rclone/rclone_test.go [hunk 4]
```diff
@@ -19,25 +19,60 @@ package rclone
...
-	createTestRCloneMockResponders(host, port, status, body)
+	createTestRCloneMockRespondersBasic(host, port, status, body)
+	return r
+}
+
+func createFakeRcloneClientForCopy(t *testing.T, status int, body string, createLocalFolder bool) *RCloneClient {
+	logger := log.New()
+	log.SetLevel(log.DebugLevel)
+	host := "rclone-server"
+	port := 5572
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply scheduler/pkg/agent/rclone/rclone_test.go:73]: None of the ports are actually used as they are either mocked or not used in the tests.

File: samples/models/error-uri-notexist.yaml (hunk 2 above)
Comment (cliveseldon) [reply samples/models/error-uri-notexist.yaml:1]: will rmeove

**Review by sakoush — APPROVED** [review]
LGTM

**Pull request merged**